flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/nodes/datastream/DataStreamCalc.scala

```diff
@@ -92,11 +92,13 @@ class DataStreamCalc(
 
     val genFunction = generator.generateFunction(
```

Comment:
If we generate a `FlatMapFunction[Row, Row]` we can use a `RetractFlatMapRunner` class to unwrap the `Row` from the `CRow` before calling the generated code.